pkgs/repo_manage/lib/issue_transfer.dart

```diff
@@ -0,0 +1,227 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:github/github.dart';
+import 'package:graphql/client.dart';
+
+import 'src/common.dart';
+
+class TransferIssuesCommand extends ReportCommand {
+  TransferIssuesCommand()
+      : super('transfer-issues',
+            'Bulk transfer issues from one repo to another.') {
+    argParser.addFlag(
+      'apply-changes',
+      negatable: false,
+      help: 'WARNING: This will transfer the issues. Please preview the changes'
+          "first by running without '--apply-changes'.",
+      defaultsTo: false,
+    );
+    argParser.addMultiOption(
+      'issue-numbers',
+      valueHelp: '1,2,3',
+      help: 'Specifiy the numbers of specific issues to transfer, otherwise'
+          ' transfers all.',
+    );
+    argParser.addOption(
+      'source-repo',
+      valueHelp: 'repo-org/repo-name',
+      help: 'The source repository for the issues to be moved from.',
+    );
+    argParser.addOption(
+      'target-repo',
+      valueHelp: 'repo-org/repo-name',
+      help: 'The target repository name where the issues will be moved to.',
+    );
+    argParser.addOption(
+      'add-label',
+      help: 'Add a label to all transferred issues.',
+      valueHelp: 'pkg:foo',
+    );
+  }
+
+  @override
+  String get invocation =>
+      '${super.invocation} --source-repo repo-org/old-repo-name --target-repo repo-org/new-repo-name --add-label pkg:old-repo-name';
+
+  @override
+  Future<int> run() async {
+    var applyChanges = argResults!['apply-changes'] as bool;
+
+    var sourceRepo = argResults!['source-repo'] as String?;
+    var targetRepo = argResults!['target-repo'] as String?;
+    if (targetRepo == null || sourceRepo == null) {
+      print('target-repo and source-repo must be specified.');
+      exit(0);
+    }
+
+    var issueNumberString = argResults!['issue-numbers'] as List<String>?;
+    var issueNumbers = issueNumberString?.map(int.parse).toList();
+    var labelName = argResults!['add-label'] as String?;
+
+    return await transferIssues(
+      RepositorySlug.full(sourceRepo),
+      RepositorySlug.full(targetRepo),
+      issueNumbers,
+      labelName,
+      applyChanges,
+    );
+  }
+
+  Future<int> transferIssues(
+    RepositorySlug sourceRepo,
+    RepositorySlug targetRepo, [
+    List<int>? issueNumbers,
+    String? labelName,
+    bool applyChanges = false,
+  ]) async {
+    if (labelName != null) {
+      print('Create label $labelName');
+      if (applyChanges) {
+        await reportRunner.github.issues.createLabel(targetRepo, labelName);
```

Comment:
Is this a no-op if the label already exists on the target repo, or will the operation throw?

Reply:
It's a no-op :)